selfdrive/car/subaru/subarucan.py

```diff
@@ -1,19 +1,36 @@
 import copy
 from cereal import car
+from selfdrive.car.subaru.values import PREGLOBAL_CARS
 
 VisualAlert = car.CarControl.HUDControl.VisualAlert
 
-def create_steering_control(packer, apply_steer, frame, steer_step):
+def subaru_preglobal_checksum(packer, values, addr):
+  dat = packer.make_can_msg(addr, 0, values)[2]
+  return (sum(dat[:7])) % 256
 
-  # counts from 0 to 15 then back to 0 + 16 for enable bit
-  idx = ((frame // steer_step) % 16)
+def create_steering_control(packer, car_fingerprint, apply_steer, frame, steer_step):
```

Comment:
Let's just split this into different functions for each platform

Reply:
done